### PR TITLE
SDIT-2237: ✨ Return list of prisoners from address lookup

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/repository/PrisonerRepository.kt [hunk 1]
```diff
@@ -12,53 +12,23 @@ import java.time.LocalDate
...
-            FORENAME_1         AS aliasFirst,
-            FORENAME_2         AS aliasMiddle
-      FROM (
-      SELECT
-        row_number()
-        OVER ( PARTITION BY PRISON_NUMBER
-        ORDER BY (IS_ALIAS) ) ROW_NUM,
-    *
-    FROM HPA.PRISONERS
+    $SELECT_CLAUSE
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/repository/PrisonerRepository.kt:18]: remove duplicated code into constants

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/repository/PrisonerRepository.kt [hunk 2]
```diff
@@ -88,4 +57,41 @@ ORDER BY IS_ALIAS, PRIMARY_SURNAME, PRIMARY_INITIAL, BIRTH_DATE, RECEPTION_DATE
...
+      FROM HPA.PRISONERS
+      WHERE PRISON_NUMBER IN (
+        SELECT DISTINCT PRISON_NUMBER FROM HPA.ADDRESS_LOOKUP WHERE CONTAINS(ADDRESS_TEXT, :addressTerms)
+      )
+  )
+  $WHERE_ORDER_CLAUSE""",
+    nativeQuery = true,
+  )
+  fun findByAddresses(addressTerms: String, pageRequest: Pageable): Page<PrisonerSearchDto>
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/repository/PrisonerRepository.kt:72]: move address lookup here as needs to return same as other searches

File: src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/AddressLookupResource.kt [hunk 3]
```diff
@@ -36,5 +39,5 @@ class AddressLookupResource(private val addressLookupService: AddressLookupServi
       ),
     ],
   )
-  fun findPrisonersWithAddresses(addressTerms: String): List<String> = addressLookupService.findPrisonersWithAddresses(addressTerms)
+  fun findPrisonersWithAddresses(addressTerms: String, pageRequest: Pageable): Page<PrisonerSearchDto> = addressLookupService.findPrisonersWithAddresses(addressTerms, pageRequest)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/historicalprisonerapi/resource/AddressLookupResource.kt:42]: should have been returning page of prisoner search results

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**